### PR TITLE
Refactor Cassandra scaler

### PR DESCRIPTION
**Pull request by rickbrouwer**
- Refactor the scaler to match the new metadata guidelines.

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Tests have been added
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Related: [5797](https://github.com/kedacore/keda/pull/5797)

### CODE REVIEW

File: pkg/scalers/cassandra_scaler.go [hunk 1]
```diff
@@ -73,117 +86,41 @@ func NewCassandraScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
...
-		meta.password = val
-	} else {
-		return nil, fmt.Errorf("no password given")
+	err = meta.Validate()
+	if err != nil {
+		return meta, err
 	}
 
-	if err = parseCassandraTLS(config, meta); err != nil {
+	err = parseCassandraTLS(&meta)
```
Comment (SpiritZhou) [inline pkg/scalers/cassandra_scaler.go:109]: Maybe this function can be moved to Validate() as well.

File: pkg/scalers/cassandra_scaler.go [hunk 2]
```diff
@@ -73,117 +86,41 @@ func NewCassandraScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
...
-	if val, ok := config.TriggerMetadata["keyspace"]; ok {
-		meta.keyspace = val
-	} else {
-		return nil, fmt.Errorf("no keyspace given")
-	}
-	if val, ok := config.AuthParams["password"]; ok {
-		meta.password = val
-	} else {
-		return nil, fmt.Errorf("no password given")
+	err = meta.Validate()
```
[the remark this answers is not in the document]
Comment (rickbrouwer) [reply pkg/scalers/cassandra_scaler.go]: Removed

File: pkg/scalers/cassandra_scaler.go (hunk 1 above)
Comment (rickbrouwer) [reply pkg/scalers/cassandra_scaler.go:108]: I moved the check which is in the `parseCassandraTLS`moved to `Validate()`

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**